providers/go-feature-flag/src/main/java/dev/openfeature/contrib/providers/gofeatureflag/api/GoFeatureFlagApi.java

```diff
@@ -0,0 +1,283 @@
+import dev.openfeature.contrib.providers.gofeatureflag.api.bean.FlagConfigApiResponse;
+import dev.openfeature.contrib.providers.gofeatureflag.api.bean.OfrepRequest;
+import dev.openfeature.contrib.providers.gofeatureflag.api.bean.OfrepResponse;
+import dev.openfeature.contrib.providers.gofeatureflag.bean.FlagConfigResponse;
+import dev.openfeature.contrib.providers.gofeatureflag.bean.GoFeatureFlagResponse;
+import dev.openfeature.contrib.providers.gofeatureflag.bean.IEvent;
+import dev.openfeature.contrib.providers.gofeatureflag.exception.FlagConfigurationEndpointNotFound;
+import dev.openfeature.contrib.providers.gofeatureflag.exception.ImpossibleToRetrieveConfiguration;
+import dev.openfeature.contrib.providers.gofeatureflag.exception.ImpossibleToSendEventsException;
+import dev.openfeature.contrib.providers.gofeatureflag.exception.InvalidEndpoint;
+import dev.openfeature.contrib.providers.gofeatureflag.exception.InvalidOptions;
+import dev.openfeature.contrib.providers.gofeatureflag.util.Const;
+import dev.openfeature.contrib.providers.gofeatureflag.validator.Validator;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.exceptions.FlagNotFoundError;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import dev.openfeature.sdk.exceptions.InvalidContextError;
+import dev.openfeature.sdk.exceptions.OpenFeatureError;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import okhttp3.ConnectionPool;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+/**
+ * GoFeatureFlagApi is the class to contact the GO Feature Flag relay proxy.
+ */
+@Slf4j
+public final class GoFeatureFlagApi {
+    /** apiKey contains the token to use while calling GO Feature Flag relay proxy. */
+    private final String apiKey;
+
+    /** httpClient is the instance of the OkHttpClient used by the provider. */
+    private final OkHttpClient httpClient;
+
+    /** parsedEndpoint is the endpoint of the GO Feature Flag relay proxy. */
+    private final HttpUrl parsedEndpoint;
+
+    /**
+     * GoFeatureFlagController is the constructor of the controller to contact the GO Feature Flag
+     * relay proxy.
+     *
+     * @param options - options to initialise the controller
+     * @throws InvalidOptions - if the options are invalid
+     */
+    @Builder
+    private GoFeatureFlagApi(final GoFeatureFlagProviderOptions options) throws InvalidOptions {
+        Validator.providerOptions(options);
+        this.apiKey = options.getApiKey();
+        this.parsedEndpoint = HttpUrl.parse(options.getEndpoint());
+        if (this.parsedEndpoint == null) {
```

Comment:
this validation can be on the options class itself, no ?

Reply:
good catch I will move it.